src/directConnect.test.ts

```diff
@@ -505,4 +513,68 @@ describe("directConnect.ts", () => {
       assert.deepStrictEqual(result, { name: "test", ssl: { enabled: true } });
     });
   });
+
+  describe("handleConnectionChange", () => {
+    let sandbox: sinon.SinonSandbox;
+    let mockDispose: sinon.SinonStub;
+    let mockShowInformationMessage: sinon.SinonStub;
+    let stubResourceManager: sinon.SinonStubbedInstance<ResourceManager>;
+
+    beforeEach(() => {
+      sandbox = sinon.createSandbox();
+
+      // Mock WebviewPanel
+      mockDispose = sandbox.stub();
+
+      // Mock window.showInformationMessage
+      mockShowInformationMessage = sandbox.stub(vscode.window, "showInformationMessage");
+
+      // Mock resource manager with required methods
+      stubResourceManager = sinon.createStubInstance(ResourceManager);
+      sandbox.stub(ResourceManager, "getInstance").returns(stubResourceManager);
+    });
+
+    afterEach(() => {
+      sandbox.restore();
+    });
+
+    it("should close the form when connection is removed", async () => {
+      // Arrange: Create a test connection
+      const testConnection: CustomConnectionSpec = TEST_DIRECT_CONNECTION_FORM_SPEC;
+
+      // Setup mock for getDirectConnections to simulate connection removed
+      (stubResourceManager.getDirectConnection as sinon.SinonStub).resolves(null);
```

Comment:
Minor: since `stubResourceManager` is a `SinonStubbedInstance`, its methods will be `SinonStub`s already
```suggestion
      stubResourceManager.getDirectConnection.resolves(null);
```